### DISCUSSION TITLE
How can I use my NFS provisioner service with prometheus-operator ?

### DISCUSSION THREAD

**Discussion by journeytosilius** [prometheus-operator/prometheus-operator#3721] [Q&A] [answered]
I have followed this article to build a configuration that spawns an NFS provision container via a service account:
https://medium.com/@myte/kubernetes-nfs-and-dynamic-nfs-provisioning-97e2afb8b4a9

This way, volumes are dynamically created from my NFS server and supposedly attached to anything requiring a persistent volume. My plan was to setup persistence for this prometheus stack deployment. How can I configure the chart or, configure the stack after installation in order for the prometheus stack to run that persistent volume claim ?

I have followed this : https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/storage.md

**Comment by coderanger** [comment]
As with the blog post you linked, you can set the `storageClassName: nfs` or whatever your NFS storage class name is. That is passed through to the underlying StatefulSet spec.

**Comment by simonpasquier** [comment, marked as answer]
You're probably aware of this but bear in mind that [Prometheus](https://prometheus.io/docs/prometheus/latest/storage/#operational-aspects) doesn't recommend using NFS for storage.

**Comment by journeytosilius** [comment]
Thanks for the information, I wasn't aware. On this case, I guess the solution for now is to just provision local storage. Is there a guide to make sure the volumes are persistent then ?
